### PR TITLE
[Model] Ovis Image Model Addition

### PR DESCRIPTION
**Pull request by divyanshsinghvi**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Adds New Model Ovis Image 7B #224 

## Test Plan

`python examples/offline_inference/qwen_image/text_to_image.py --model AIDC-AI/Ovis-Image-7B --num_inference_steps 15 --seed 0 --height 256 --width 256`

## Test Result

qwen_image_output

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by divyanshsinghvi** [comment]
@hsliuustc0106 @Isotr0py  Any good way I can debug it? Right now I ran the test command 
`python examples/offline_inference/qwen_image/text_to_image.py --model AIDC-AI/Ovis-Image-7B`
but it's still outputting noisy image. 

Any tools I can use to debug it?

**Comment by divyanshsinghvi** [comment]
@Codex review

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+
+        if not return_dict:
+            return (output,)
+
+        return Transformer2DModelOutput(sample=output)
+
+
+def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]) -> set[str]:
+    stacked_params_mapping = [
+        # self attn
```
[the remark this answers is not in the document]
Comment (divyanshsinghvi) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]: I wonder why it didn't fail here? I should probably add a test around this.

**Comment by divyanshsinghvi** [comment]
> @hsliuustc0106 @Isotr0py Any good way I can debug it? Right now I ran the test command `python examples/offline_inference/qwen_image/text_to_image.py --model AIDC-AI/Ovis-Image-7B` but it's still outputting noisy image.
> 
> Any tools I can use to debug it?

Ok after fixing the weight loading correctly, its working now. But still would be interested to know what would be best ways to debug these issues?

**Comment by hsliuustc0106** [comment]
it looks like the scripts under offline_inference/qwen_image/ folder not only works for qwen-image but works for all t2i and editing jobs, could you please help to submit an additional PR to refactor the folders offline_inference/qwen_image/.. -> offline_inference/text_to_image/..

**Comment by hsliuustc0106** [comment]
> it looks like the scripts under offline_inference/qwen_image/ folder not only works for qwen-image but works for all t2i and editing jobs, could you please help to submit an additional PR to refactor the folders offline_inference/qwen_image/.. -> offline_inference/text_to_image/..

@SamitHuang @ZJY0516 WDYT

**Comment by ZJY0516** [comment]
> it looks like the scripts under offline_inference/qwen_image/ folder not only works for qwen-image but works for all t2i and editing jobs, could you please help to submit an additional PR to refactor the folders offline_inference/qwen_image/.. -> offline_inference/text_to_image/..

I think it's reasonable

**Comment by divyanshsinghvi** [comment]
@hsliuustc0106 @ZJY0516 Will raise a separate PR for the refactor. 
Anything else needed for this model to be added?

**Comment by ZJY0516** [comment]
> Anything else needed for this model to be added?

Please update supported model list in doc and add a test for this model in CI

**Comment by SamitHuang** [comment]
> > it looks like the scripts under offline_inference/qwen_image/ folder not only works for qwen-image but works for all t2i and editing jobs, could you please help to submit an additional PR to refactor the folders offline_inference/qwen_image/.. -> offline_inference/text_to_image/..
> 
> @SamitHuang @ZJY0516 WDYT

agreed

**Comment by hsliuustc0106** [comment]
for the CI, please refer to #257

**Comment by david6666666** [comment]
> > > it looks like the scripts under offline_inference/qwen_image/ folder not only works for qwen-image but works for all t2i and editing jobs, could you please help to submit an additional PR to refactor the folders offline_inference/qwen_image/.. -> offline_inference/text_to_image/..
> > 
> > 
> > @SamitHuang @ZJY0516 WDYT
> 
> agreed

#274 will update

File: examples/offline_inference/qwen3_omni/end2end.py [hunk 2]
```diff
@@ -12,13 +12,13 @@
...
 import soundfile as sf
 from PIL import Image
-from vllm import SamplingParams
 from vllm.assets.audio import AudioAsset
 from vllm.assets.image import ImageAsset
 from vllm.assets.video import VideoAsset, video_to_ndarrays
 from vllm.multimodal.image import convert_image_mode
 from vllm.utils import FlexibleArgumentParser
 
+from vllm import SamplingParams
```
Comment (hsliuustc0106) [inline examples/offline_inference/qwen3_omni/end2end.py]: why we need to change the import line?

Comment (divyanshsinghvi) [reply examples/offline_inference/qwen3_omni/end2end.py]: Weirdly my pre-commit hook is rearranging it. Checking it.

**Comment by divyanshsinghvi** [comment]
@ZJY0516 @hsliuustc0106 tried adding CI, if you have more ideas on test, happy to add more.

**Comment by ZJY0516** [comment]
@SamitHuang What are your thoughts on the added tests?

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+                    num_attention_heads=num_attention_heads,
+                    attention_head_dim=attention_head_dim,
+                )
+                for _ in range(num_single_layers)
+            ]
+        )
+        self.norm_out = AdaLayerNormContinuous(self.inner_dim, self.inner_dim, elementwise_affine=False, eps=1e-6)
+        self.proj_out = nn.Linear(self.inner_dim, patch_size * patch_size * self.out_channels, bias=True)
+
+        self.gradient_checkpointing = False
```
Comment (Isotr0py) [inline vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]:
```suggestion
```
We should remove gradient checkpointing codes because it's not used by inference runtime.

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+            if torch.is_grad_enabled() and self.gradient_checkpointing:
+                encoder_hidden_states, hidden_states = self._gradient_checkpointing_func(
+                    block,
+                    hidden_states,
+                    encoder_hidden_states,
+                    temb,
+                    image_rotary_emb,
+                )
+
+            else:
```
Comment (Isotr0py) [inline vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]:
```suggestion
```
Ditto

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
```
Comment (Isotr0py) [inline vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]:
```suggestion
```
Ditto

Comment (divyanshsinghvi) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]: Fixed

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py (hunk 4 above)
Comment (divyanshsinghvi) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]: Fixed

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py (hunk 3 above)
Comment (divyanshsinghvi) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py]: Fixed

**Comment by divyanshsinghvi** [comment]
cc: @SamitHuang 

What do you think about CI tests? Any other improvements that can be made?

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Iterable
+from typing import Any, Optional, Union
+
+import torch
+import torch.nn as nn
+from diffusers.models.attention import FeedForward
```
Comment (ZJY0516) [inline vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py:23]: Could you please implement FeedForward in this file? We may want to extend it to support tp later.

**Comment by ZJY0516** [comment]
Let's try to run CI first

**Comment by hsliuustc0106** [comment]
please fix the precommits and let's get it merged asap :)

File: vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py (hunk 6 above)
Comment (divyanshsinghvi) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py:23]: I will raise a separate PR for this?

Comment (ZJY0516) [reply vllm_omni/diffusion/models/ovis_image/ovis_image_transformer.py:23]: No problem

**Comment by divyanshsinghvi** [comment]
> please fix the precommits and let's get it merged asap :)

Fixed.

File: tests/e2e/offline_inference/test_ovis_image.py [hunk 7]
```diff
@@ -0,0 +1,278 @@
+"""
```
Comment (david6666666) [inline tests/e2e/offline_inference/test_ovis_image.py:1]: this file should under vllm-omni/tests/e2e/offline_inference

File: .buildkite/pipeline.yml [hunk 8]
```diff
@@ -28,6 +28,7 @@ steps:
     depends_on: image-build
     commands:
       - pytest -s -v tests/e2e/offline_inference/test_t2i_model.py
+      - pytest -s -v tests/single_stage/test_ovis_image.py
```
Comment (david6666666) [inline .buildkite/pipeline.yml]: ditto

Comment (divyanshsinghvi) [reply .buildkite/pipeline.yml]: done

File: tests/e2e/offline_inference/test_ovis_image.py (hunk 7 above)
Comment (divyanshsinghvi) [reply tests/e2e/offline_inference/test_ovis_image.py:1]: refactored.
